### PR TITLE
Fixed injection in message-order MCM mode

### PR DESCRIPTION
**Pull request by jhh67**
Modified message-order MCM mode to not use fi_inject* functions as they do not generate a completion event. See
the comment in` comm-ofi.c` for details. Also cleaned up transmit context initialization and refactored several
functions to have similar control flow. 

Signed-off-by: John H. Hartman <jhh67@users.noreply.github.com>

### CODE REVIEW

File: runtime/src/comm/ofi/comm-ofi.c [hunk 1]
```diff
@@ -6086,20 +6030,14 @@ chpl_comm_nb_handle_t rmaPutFn_msgOrdFence(void* myAddr, void* mrDesc,
     //
     flags = FI_INJECT;
   }
-  if (tcip->bound && bitmapTest(tcip->amoVisBitmap, node)) {
+  if (bitmapTest(tcip->amoVisBitmap, node)) {
```
Comment (jabraham17) [inline runtime/src/comm/ofi/comm-ofi.c:6033]: why is `tcip->bound` no longer checked here?

**Review by jabraham17 — APPROVED** [review]

File: runtime/src/comm/ofi/comm-ofi.c (hunk 1 above)
Comment (jhh67) [reply runtime/src/comm/ofi/comm-ofi.c:6033]: It's not needed and kind of confused me when I first saw it. `bitmapTest` will only return `true` if `tcip->bound` is `true`, and `tcip->bound` isn't checked in other similar conditionals so I removed it.

**Comment by bradcray** [comment]
@jhh67 / @jabraham17 : Any idea whether this will have any impact on performance?

**Comment by jhh67** [comment]
It's more of a correctness issue than a performance issue, as hangs are possible if completion events are not generated. There is an outstanding issue with injection captured in https://github.com/Cray/chapel-private/issues/6525; once that is resolved we can track the EX performance tests to see if there is a performance impact.

**Pull request merged**